transform/apply.go

```diff
@@ -61,6 +80,7 @@ func (t *Transformer) Apply(
 	wait bool,
 	scriptOut io.Writer,
 	secrets map[string]string,
+	mode Mode,
```

Comment:
Given that we're switching away from an "enumeration" (we don't  have true enums in go), I'd like a stronger mechanism for checking that we're using this correctly. Given that we're only talking about two possible modes, the right way to approach this is with a set of two exported methods that both call the same unexpected method with different `RunMode` strings:

```go

// Apply applies the transform script to a target dataset
func (t *Transformer) Apply(
	ctx context.Context,
	target *dataset.Dataset,
	runID string,
	wait bool,
	scriptOut io.Writer,
	secrets map[string]string,
) error {
	return t.apply(ctx, target, runID, wait, scriptOut, secrets, RMApply)
}

// Commit applies the transform script to a target dataset, associating all 
// events with the "commit" RunMode
func (t *Transformer) Commit(
	ctx context.Context,
	target *dataset.Dataset,
	runID string,
	wait bool,
	scriptOut io.Writer,
	secrets map[string]string,
) error {
	return t.apply(ctx, target, runID, wait, scriptOut, secrets, RMCommit)
}

func (t *Transformer) apply(
	ctx context.Context,
	target *dataset.Dataset,
	runID string,
	wait bool,
	scriptOut io.Writer,
	secrets map[string]string,
	runMode string,
) error {
	log.Debugw("applying transform", "runID", runID, "wait", wait)

	if target.Transform == nil {
		return errors.New("apply requires a transform component")
	}
      // ...
```

here the old `Apply` now becomes an unexported function that both reference, and within `lib/datasets.go/Save` we'd call `transformer.Commit` instead of apply. In the future we can invert this & have the transformer call save by adding to the `Commit` method.

Adding this second method to `transformer` means we should also rename this file from `apply.go` to `transform.go`

Reply:
perf, love it